utils/frame/remote-externalities/src/lib.rs

```diff
@@ -630,18 +641,27 @@ impl<B: BlockT + DeserializeOwned> Builder<B> {
 
 	pub(crate) async fn init_remote_client(&mut self) -> Result<(), &'static str> {
 		let mut online = self.as_online_mut();
-		log::debug!(target: LOG_TARGET, "initializing remote client to {:?}", online.transport.uri);
 
-		// First, initialize the ws client.
-		let ws_client = WsClientBuilder::default()
-			.max_request_body_size(u32::MAX)
-			.build(&online.transport.uri)
-			.await
-			.map_err(|e| {
-				log::error!(target: LOG_TARGET, "error: {:?}", e);
-				"failed to build ws client"
-			})?;
-		online.transport.client = Some(ws_client);
+		let maybe_transport = if let Transport::Uri(uri) = &online.transport {
+			log::debug!(target: LOG_TARGET, "initializing remote client to {:?}", uri);
+
+			// First, initialize the ws client.
+			let ws_client = WsClientBuilder::default()
+				.max_request_body_size(u32::MAX)
+				.build(&uri)
+				.await
+				.map_err(|e| {
+					log::error!(target: LOG_TARGET, "error: {:?}", e);
+					"failed to build ws client"
+				})?;
+			Some(Transport::RemoteClient(Arc::new(ws_client)))
+		} else {
+			None
+		};
+
+		if let Some(transport) = maybe_transport {
+			online.transport = transport;
+		}
```

Comment:
```suggestion
		online.transport = if let Transport::Uri(uri) = &online.transport {
			log::debug!(target: LOG_TARGET, "initializing remote client to {:?}", uri);

			// First, initialize the ws client.
			let ws_client = WsClientBuilder::default()
				.max_request_body_size(u32::MAX)
				.build(&uri)
				.await
				.map_err(|e| {
					log::error!(target: LOG_TARGET, "error: {:?}", e);
					"failed to build ws client"
				})?;
			Some(Transport::RemoteClient(Arc::new(ws_client)))
		} else {
			None
		};
```

are the same? maybe I am missing something

Reply:
yeah except the option stuff is not needed anymore, nice catch.

Saved us a few lines

Reply:
But `online.transport` isn't an Option?

We could add some `map_uri` function to `Transport`, then we would also not require the extra option.

Reply:
yeah, you are right ty.

good to get rid of that extra mess.